### ISSUE TITLE
Tasks

### ISSUE THREAD

**Issue by dbosk** [dbosk/crocus#133]
This is an issue version of `2dosAug29.txt`, so we can remove that file.

- [x] Remove `2dosAug29.txt` from repo.

Collected explanations, tasks for *Daniel*:

- [x] - Daniel: fix statement "we require public-key encryption for discrete logarithms" (end of DB building block) and give reasons
- [x] - Daniel: make sure DBS is explained, including what the names mean
- [ ] - Daniel: explain how DoS would work [I'll procrastinate this one, I have no idea and maybe we should rewrite that anyway?]
- [x] - Daniel: explain "moreover, if they provided individual verifiability, they would likely violate any privacy requirements"
- [x] - Daniel: "before and after the protest, we assume [...] global connectivity" why before? Needed to check current head of blockchain? Need to explain what the participants do throughout.
- [x] - Daniel: redo figure 4

*Sonja*:

tasks:

- [x] collect items from annotations
- [x] list of adversaries, a1-a4, informal
- [x] - little edits throughout
- [x] - system and adv. model, abstract and concrete, move parts from discussion to model
- [x] - privacy properties: back to p1 and p2
- [x] added comment re: more differences between PROPS and crocus? (related work location proofs)
- [ ] - adversary models in e-voting, which properties do they achieve under which adversary model
- [x] - performance (sub) section
- [x] - verification subsection in DB Schnorr section

explanations:

- [x] - protest as stand-in for event
- [x] - different counts possible but if counting method is given, can be verified
- [x] x roles: presence could mean participant or bystander, participant could mean protester or witness (e.g. journalist) or both
- [x] x added comment: first GENERAL DB AC
- [x] x added comment: related work: platin.io in location proofs, details unknown but roughly relying on witnesses and graph theory (unique big cluster, assumption of honest majority)
- [x] x added comment IF resistance follows from Schnorr, our changes don't break Schnorr because ...
- [x] x MF far-away (from verifier)
- [ ] - name verifier impersonating prover?
- [x] x malicious verifier (only concerning impersonating prover) not handled in most DB schemes due to being symmetric. The few public-key schemes implicitly prevent this. symmetric (shared-secret) DB schemes assume an honest verifier (or honest-but-curious that tries to track) 
- [x] - verifier impersonating prover: vulnerability we introduce against public-key DB schemes, mitigated by limiting size of l (bits of other s). Trade-off: want long l for reducing success of guessing correct sequence of challenge bits, want short l for reducing success of guessing the other s and finding alpha
- [x] x same as proprox reason? needed because of type of AC (CL04) for PRF for deriving identifiers from ID with unlinkability to the original ID but with proof of validity of this derivation
- [x] x proprox good in principle, but need specifics (due to CL04)  that are not provided 
- [x] - attacks on BC by [10]
- [x] x why two challenges (change from Schnorr): knowing the responses to both means learning alpha (TF), authentication bundled into distance bounding (additional change from BC-Schnorr) (DH), both (two challenges, auth in DB) reduce success of MF according to number of repetitions.
- [x] - performance: important during protest due to resource-constrained devices (battery, computational power, storage) and limited network capacity. Less important after, at other locations with plugged-in devices
- [x] - verification with tamarin: there is no BC Schnorr in Tamarin,  [10] assumed equivalence with Fiat-Shamir model. Not straightforward to implement Schnorr in Tamarin due to a lack of expressiveness for arithmetic needed to complement Diffie-Hellman. If not done now, left for future work.
- [x] - pid is always the protester's pseudonym, never the witness's
- [x] - Jane can use a group signature (?) for trusted journalists, clarify in other actors, "reveals her identity" at least proof ok knowledge of secret key for (group of) trusted journalist
- [x] - add to FireChat that we likely have higher requirements on capacity and withstanding interference as everyone runs the witnessing with others continuously; messages are presumably less frequent.
- [x] - threshold based on contacts irrelevant for online forums with large number of participants (from the same country)
- [x] - add "at least" to blockchain times as threshold is determined at count and verification, not yet known at upload.
- [x] - conclusion: have anonymity and unlinkability unless adversary gets private key/ID or has side information (also needs private key for proof of cause, otherwise does not learn more than without crocus)
- [ ] - effect of hiding cid from W: to enable hiding in the wrong crowd and still be counted for the right, i.e. to avoid discrimination of refusal to witness and detection by Grace when she acts as a witness in pro-government counter-protest
- [ ] - effect of hiding cid from witness plus blinding pid: sever link of wid that Grace has at protest (and side information) from what she sees when protester uploads their part, as there is no wid uploaded. A mandatory Jane enables blinding of pid because Jane gives a non-interactive proof that her sig(blinded pid, l, t) matches her known public key.
- [ ] - naming: P for protester and protest, change one?
- [ ] - naming: wid for testimony, witness in the cryptographic sense
- [ ] - p1 necessary but not sufficient for p2 = p2 implies p1
- [ ] - a3 can be thwarted by bulk uploads

*Henri*, performance:

- [ ] - 80 bits (as in number of runs) reasonable for DB because of time limitation, compromise with risk of loss of messages. Reference?
- [ ] - re-run experiment with q,g=2048, k=10 (or 1, see Camenisch), 80, 128, 512, 1024 runs
- [x] - separate pre-computation (done only once) from rest of setup phase
- [x] - what to do with Anonpass numbers, which to plug into which formula, which conclusions to draw
- [ ] - twin Schnorr https://www.hindawi.com/journals/tswj/2015/237514/

list of adversaries (informal)
a1: adversary with only in-system data available (local and on ledger)
in practice: on own phone or blockchain

a2: a1 plus with side information (real ID) 
in practice: takes picture with time stamp, uses face recognition that yields real identity

a3: global passive (NW) or local passive (storage)
in practice: transcripts of all traffic, access to a miner that receives crocus data, can put together IP address and protest data

a4: adversary with access to participant's secret key
in practice: can read keys from the participant's mobile phone, by coercion/subpoena
